blaze/expr/arithmetic.py

```diff
@@ -50,6 +50,55 @@ def name(o):
         return None
 
 
+class Op(ElemWise):
+    def __init__(self, ops):
+        if not isinstance(ops, (list, tuple,)):
```

Comment:
This could also be generalized to an operator with N arguments, and that would simplify some of the code below. For example,

```python
class Op(ElemWise):
    __slots__ = 'operands',
    __inputs__ = 'operands',
    def __init__(self, operands):
        self._operands = operands
```

Then, put the different `__str__` implementations in the subclass. Generally, the parent class shouldn't know about specific details of child classes.